common/tools/dev-tool/src/commands/samples/checkNodeVersions.ts

```diff
@@ -0,0 +1,306 @@
+    cwd: dockerContextDirectory
+  });
+  log.info(`Started running the docker container ${dockerContainerName}`);
+  dockerContainerRunProcess.stdout.on("data", stdoutListener);
+  dockerContainerRunProcess.stderr.on("data", stderrListener);
+  const exitCode = await new Promise((resolve, reject) => {
+    dockerContainerRunProcess.on("exit", resolve);
+    dockerContainerRunProcess.on("error", reject);
+  });
+  if (exitCode === 0) {
+    log.info(`Docker container ${dockerContainerName} finished running`);
+  } else {
+    log.error(`Docker container ${dockerContainerName} encountered an error`);
+  }
+}
+
+export const commandInfo = makeCommandInfo(
+  "check-node-versions",
+  "execute samples with different node versions, typically in preparation for release",
+  {
+    "artifact-path": {
+      kind: "string",
+      description: "Path to the downloaded artifact built by the release pipeline"
+    },
+    directory: {
+      kind: "string",
+      description: "Base dir, default is process.cwd()",
+      default: process.cwd()
+    },
+    "node-versions": {
+      kind: "string",
+      description: "A comma separated list of node versions to use",
+      default: "8,10,12"
+    },
+    "context-directory-path": {
+      kind: "string",
+      description: "Absolute path to a directory used for mounting inside docker containers",
+      default: ""
+    },
+    "keep-docker-context": {
+      kind: "boolean",
+      description: "Boolean to indicate whether to keep the current docker context directory",
+      default: false
+    },
+    "log-in-file": {
+      kind: "boolean",
+      description:
+        "Boolean to indicate whether to save the the stdout and sterr for npm commands to the log.txt log file",
+      default: true
+    },
+    "use-existing-docker-containers": {
+      kind: "boolean",
+      description: "Boolean to indicate whether to use existing docker containers if any",
+      default: false
+    },
+    "keep-docker-containers": {
+      kind: "boolean",
+      description: "Boolean to indicate whether to keep docker containers",
+      default: false
+    },
+    "keep-docker-images": {
+      kind: "boolean",
+      description: "Boolean to indicate whether to keep the downloaded docker images",
+      default: false
+    }
+  }
+);
+
+export default leafCommand(commandInfo, async (options) => {
+  const nodeVersions = options["node-versions"]?.split(",");
+  const dockerContextDirectory: string =
+    options["context-directory-path"] === ""
+      ? await fs.mkdtemp(path.join(os.tmpdir(), "context"))
+      : options["context-directory-path"];
+  const pkg = await resolveProject(options.directory);
+  const samplesPath = path.join(pkg.path, "samples");
+  const envFilePath = path.join(pkg.path, ".env");
+  const keepDockerContextDirectory = options["keep-docker-context"];
+  const dockerImageNames = nodeVersions.map((version: string) => `node:${version}-alpine`);
```

Comment:
Nice, so this must have worked okay for you then? I wasn't sure if you were still going with the nvm approach or not.

Reply:
Yup, my concern was the I/O performance on mounted directories but I realized most of our samples do not do that anyway. Also, running an already built image is much faster.